src/utils/fetch.ts

```diff
@@ -10,9 +9,13 @@ export const getFetch = (config: PineconeConfiguration) => {
     // scope, use that. This should prevent confusing failures in
     // nextjs projects where @vercel/fetch is mandated and
     // other implementations are stubbed out.
+    console.log(
+      'Failed to find any user-provided fetch implementation. Searching for global fetch implementation...'
+    );
     return global.fetch;
   } else {
-    // Use ponyfill as last resort
-    return crossFetch;
+    console.error(
+      'No global or user-provided fetch implementations found. Please supply a fetch implementation.'
+    );
```

Comment:
Do we want to actually throw here? I think this will just log an error to the console without interrupting control flow. If there's no available fetch implementation they wont be able to do anything with the client anyways, so maybe we should more forcefully exit here?

Reply:
Good point, I'll turn it into a throw!

Reply:
(ended up adding some unit tests to confirm everything, too)